Content.Client/Inventory/ClientInventorySystem.cs

```diff
@@ -25,39 +57,276 @@ public override void Initialize()
+                    }
                 }
             }
         }
 
         public override void Shutdown()
         {
             CommandBinds.Unregister<ClientInventorySystem>();
+            _config.UnsubValueChanged(CCVars.HudTheme, UpdateHudTheme);
             base.Shutdown();
         }
 
+        private void OnInit(EntityUid uid, ClientInventoryComponent component, ComponentInit args)
+        {
+            _clothingSystem.InitClothing(uid, component);
+
+            if (!TryGetUIElements(uid, out var window, out var bottomLeft, out var bottomRight, out var topQuick,
+                    component))
+                return;
+
+            component.InventoryWindow = window;
+            component.BottomLeftButtons = bottomLeft;
+            component.BottomRightButtons = bottomRight;
+            component.TopQuickButtons = topQuick;
+        }
+
+        private void HoverInSlotButton(EntityUid uid, string slot, ItemSlotButton button, InventoryComponent? inventoryComponent = null, SharedHandsComponent? hands = null)
+        {
+            if (!Resolve(uid, ref inventoryComponent))
+                return;
+
+            if (!Resolve(uid, ref hands, false))
+                return;
+
+            if (!hands.TryGetActiveHeldEntity(out var heldEntity))
+                return;
+
+            _itemSlotManager.HoverInSlot(button, heldEntity,
+                CanEquip(uid, heldEntity, slot, out _, inventory: inventoryComponent));
+        }
+
+        private void HandleSlotButtonPressed(EntityUid uid, string slot, ItemSlotButton button,
+            GUIBoundKeyEventArgs args)
+        {
+            if (TryGetSlotEntity(uid, slot, out var itemUid))
+            {
+                if (!_itemSlotManager.OnButtonPressed(args, itemUid.Value) && args.Function == EngineKeyFunctions.UIClick)
+                {
+                    RaiseNetworkEvent(new UseSlotNetworkMessage(uid, slot));
+                }
+                return;
+            }
+
+            if (args.Function != EngineKeyFunctions.UIClick) return;
+            TryEquipActiveHandTo(uid, slot);
+        }
+
+        private bool TryGetUIElements(EntityUid uid, [NotNullWhen(true)] out SS14Window? invWindow,
+            [NotNullWhen(true)] out Control? invBottomLeft, [NotNullWhen(true)] out Control? invBottomRight,
+            [NotNullWhen(true)] out Control? invTopQuick, ClientInventoryComponent? component = null)
+        {
+            invWindow = null;
+            invBottomLeft = null;
+            invBottomRight = null;
+            invTopQuick = null;
+
+            if (!Resolve(uid, ref component))
+                return false;
+
+            if(!_prototypeManager.TryIndex<InventoryTemplatePrototype>(component.TemplateId, out var template))
+                return false;
+
+            if (!_uiGenerateDelegates.TryGetValue(component.TemplateId, out var genfunc))
+            {
+                genfunc = (entityUid, list) =>
```

Comment:
_uiGenerateDelegates is private and never seems to have any values added to it? Is it supposed to cache the genfunc you create here?

But even if that's the case, why not just make genfunc an actual function that also includes the TemplateId as an argument? Why do they need to be stored in a dictionary?

Reply:
1) yep, mb, never updated it
2) that'd mean i'd have to precompile a function for *every* template, so i'd rather create & cache as it comes along